src/components/layer-controller/LayerControllerSubscriber.js

```diff
@@ -32,21 +188,55 @@ function LayerControllerSubscriber(props) {
     removeGridComponent,
     theme,
     title = 'Spatial Layers',
+    disable3D,
   } = props;
 
   const loaders = useLoaders();
 
   // Get "props" from the coordination space.
-  const [{
-    dataset,
-    spatialRasterLayers: rasterLayers,
-    spatialCellsLayer: cellsLayer,
-    spatialMoleculesLayer: moleculesLayer,
-  }, {
-    setSpatialRasterLayers: setRasterLayers,
-    setSpatialCellsLayer: setCellsLayer,
-    setSpatialMoleculesLayer: setMoleculesLayer,
-  }] = useCoordination(COMPONENT_COORDINATION_TYPES.layerController, coordinationScopes);
+  const [
+    {
+      dataset,
+      spatialRasterLayers: rasterLayers,
+      spatialCellsLayer: cellsLayer,
+      spatialMoleculesLayer: moleculesLayer,
+    },
+    {
+      setSpatialRasterLayers: setRasterLayers,
+      setSpatialCellsLayer: setCellsLayer,
+      setSpatialMoleculesLayer: setMoleculesLayer,
+      setSpatialTargetX: setTargetX,
+      setSpatialTargetY: setTargetY,
+      setSpatialTargetZ: setTargetZ,
+      setSpatialRotationX: setRotationX,
+      setSpatialRotationOrbit: setRotationOrbit,
+      setSpatialZoom: setZoom,
+    },
+  ] = useCoordination(
+    COMPONENT_COORDINATION_TYPES.layerController,
+    coordinationScopes,
+  );
+
+  const [
+    {
+      rasterLayersCallbacks,
+      rasterLayersIsChannelLoading: areLoadingRasterChannnels,
+    },
+    {
+      setRasterLayersCallbacks,
+      setRasterLayersIsChannelLoading: setAreLoadingRasterChannnels,
+    },
+  ] = useAuxiliaryCoordination(
+    COMPONENT_COORDINATION_TYPES.layerController,
+    coordinationScopes,
+  );
+  const spatialDims = useViewConfigStore(state => state.viewConfig.layout
+    .filter(l => l.component === 'spatial')
+    .find(
+      l => l.coordinationScopes.spatialRasterLayers
+          === coordinationScopes.spatialRasterLayers,
+    ));
```

Comment:
I don't like the idea of doing low-level things with `useViewConfigStore` inside a component. 

Can you either make a new hook like `useComponentGridDimensions` in `src/app/state/hooks.js` which takes the component `uuid` and returns the current dimensions from the layout 

_or_  alternatively we could always pass the grid layout dimensions to the subscriber components around line 81 of `src/app/vitessce-grid-layout/VitessceGridLayout.js`?

Reply:
Done! New hook added.